plugins/dfu/dfu-device.c

```diff
@@ -1365,7 +1366,10 @@ dfu_device_attach (FuDevice *device, GError **error)
 		return FALSE;
 
 	/* normal DFU mode just needs a bus reset */
-	if (!dfu_target_attach (target, error))
+	if (fu_device_has_custom_flag (device, "no-bus-reset-attach") &&
+	    dfu_device_has_attribute (self, DFU_DEVICE_ATTRIBUTE_WILL_DETACH))
```

Comment:
do we need the quirk? It seems that we should just look at `DFU_DEVICE_ATTRIBUTE_WILL_DETACH` for this behaviour for a DFU compliant device.

Reply:
I think it would be better to keep this quirk. Maybe some other existing DFU capable devices still rely on this bus reset to attach. Removing this quirk may have impact on them.